File: scripts/remove_bg.py
import sys
from PIL import Image
import os

def remove_checkerboard(img_path):
    if not os.path.exists(img_path):
        return
        
    img = Image.open(img_path).convert("RGBA")
    data = img.getdata()
    
    new_data = []
    for item in data:
        r, g, b, a = item
        # Checkerboard is usually white (255,255,255) and light grey (around 204,204,204)
        # If the pixel is very unsaturated (grey) and light, make it transparent
        if r > 180 and g > 180 and b > 180 and abs(r-g) < 15 and abs(r-b) < 15 and abs(g-b) < 15:
            new_data.append((255, 255, 255, 0))
        else:
            # Also soften edges by making slightly darker greys semi-transparent
            if r > 150 and g > 150 and b > 150 and abs(r-g) < 20 and abs(r-b) < 20 and abs(g-b) < 20:
                 new_data.append((r, g, b, int(a * 0.5)))
            else:
                 new_data.append(item)
            
    img.putdata(new_data)
    img.save(img_path)
    print(f"Processed {img_path}")

try:
    remove_checkerboard('public/assets/gift-box.png')
    remove_checkerboard('public/assets/teddy-bear.png')
    remove_checkerboard('public/assets/two-cats.png')
    remove_checkerboard('public/assets/envelope.png')
    remove_checkerboard('public/assets/parisian-lamp.png')
except Exception as e:
    print(e)

File: scripts/remove-bg.js
const Jimp = require('jimp');
const path = require('path');

const files = ['gift-box.png', 'teddy-bear.png', 'two-cats.png', 'envelope.png'];
const assetsDir = path.join(__dirname, 'public', 'assets');

async function removeBackground() {
  for (const file of files) {
    try {
      const imagePath = path.join(assetsDir, file);
      const image = await Jimp.read(imagePath);
      
      image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
        const r = this.bitmap.data[idx + 0];
        const g = this.bitmap.data[idx + 1];
        const b = this.bitmap.data[idx + 2];
        const a = this.bitmap.data[idx + 3];
        
        // Checkerboard colors are usually around pure white (255) and light grey (204)
        // Check if the pixel is bright and grey-ish
        const isBright = r > 180 && g > 180 && b > 180;
        const isGreyish = Math.abs(r - g) < 20 && Math.abs(r - b) < 20 && Math.abs(g - b) < 20;
        
        if (isBright && isGreyish) {
          // Make it completely transparent
          this.bitmap.data[idx + 3] = 0;
        } else if (r > 150 && g > 150 && b > 150 && Math.abs(r - g) < 25 && Math.abs(r - b) < 25) {
          // Edge smoothing: semi-transparent for slightly darker greys
          this.bitmap.data[idx + 3] = Math.floor(a * 0.4);
        }
      });
      
      await image.writeAsync(imagePath);
      console.log(`Processed ${file}`);
    } catch (err) {
      console.error(`Failed to process ${file}:`, err);
    }
  }
}

removeBackground();
